app/assets/javascripts/workarea/storefront/paypal/modules/paypal.js

```diff
@@ -0,0 +1,57 @@
+/**
+ *
+ * Reliably ensure that the PayPal SDK is either loaded or will
+ * be loading.
+ *
+ * @namespace WORKAREA.paypal
+ */
+WORKAREA.registerModule('paypal', (function () {
+    'use strict';
+
+    var handlers = [],
+        attempts = 0,
+
+        /**
+         * Define a function that will be called when the paypal SDK has
+         * loaded.
+         *
+         * @method
+         * @name ready
+         * @param function handler - A function to be called when PayPal
+         *                           is available.
+         * @memberof WORKAREA.paypal
+         */
+        ready = function(handler) {
+            handlers.push(handler);
+        },
+
+        /**
+         * Continuously check for whether the Paypal SDK has loaded
+         * every 5 seconds, and call function handlers defined in
+         * `ready()` when the `window.paypal` object is available. This
+         * will only occur if the `<script>` tag containing the PayPal
+         * SDK is detected to be on the page.
+         *
+         * @method
+         * @name init
+         * @memberof WORKAREA.paypal
+         */
+        init = function($scope) {
+            var $sdk = $('script[data-partner-attribution-id]'),
+                paypalLoaded = window.paypal !== undefined,
+                paypalExpectedToLoad = !_.isEmpty($sdk);
+
+            attempts += 1;
+
+            if (!paypalLoaded && paypalExpectedToLoad && attempts <= 20) {
+                window.setTimeout(init, 2000);
+            } else if (paypalLoaded) {
+                handlers.forEach(function(handler) { handler($scope); });
```

Comment:
we're now passing the `$scope` thru here so other paypal modules can use `WORKAREA.paypal.ready()` as a replacement for their own `init()` functions.